core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java

```diff
@@ -495,20 +495,26 @@ private static String formatFragment(
                             stageStats.getOutputDataSize()));
             Optional<TDigestHistogram> outputBufferUtilization = stageInfo.get().getStageStats().getOutputBufferUtilization();
             if (verbose && outputBufferUtilization.isPresent()) {
+                TDigestHistogram utilization = outputBufferUtilization.get();
                 builder.append(indentString(1))
                         .append(format("Output buffer active time: %s, buffer utilization distribution (%%): {p01=%s, p05=%s, p10=%s, p25=%s, p50=%s, p75=%s, p90=%s, p95=%s, p99=%s, max=%s}\n",
-                                succinctNanos(outputBufferUtilization.get().getTotal()),
+                                succinctNanos(utilization.getTotal()),
                                 // scale ratio to percentages
-                                formatDouble(outputBufferUtilization.get().getP01() * 100),
-                                formatDouble(outputBufferUtilization.get().getP05() * 100),
-                                formatDouble(outputBufferUtilization.get().getP10() * 100),
-                                formatDouble(outputBufferUtilization.get().getP25() * 100),
-                                formatDouble(outputBufferUtilization.get().getP50() * 100),
-                                formatDouble(outputBufferUtilization.get().getP75() * 100),
-                                formatDouble(outputBufferUtilization.get().getP90() * 100),
-                                formatDouble(outputBufferUtilization.get().getP95() * 100),
-                                formatDouble(outputBufferUtilization.get().getP99() * 100),
-                                formatDouble(outputBufferUtilization.get().getMax() * 100)));
+                                formatDouble(utilization.getP01() * 100),
+                                formatDouble(utilization.getP05() * 100),
+                                formatDouble(utilization.getP10() * 100),
+                                formatDouble(utilization.getP25() * 100),
+                                formatDouble(utilization.getP50() * 100),
+                                formatDouble(utilization.getP75() * 100),
+                                formatDouble(utilization.getP90() * 100),
+                                formatDouble(utilization.getP95() * 100),
+                                formatDouble(utilization.getP99() * 100),
+                                formatDouble(utilization.getMax() * 100)));
+                if (utilization.getPercentile(50) > 1.0) {
```

Comment:
Why do you need to print it separately instead of just reading the P50 value ?
Also, consider if you really need all the Ps from 1 to max, it would be easier to read with fewer buckets printed.

Reply:
> Why do you need to print it separately instead of just reading the P50 value ?

This is a hint for non-power user and also for us to be able quickly glimpse over query plan to find potentially problematic stages.

> Also, consider if you really need all the Ps from 1 to max, it would be easier to read with fewer buckets printed.

Note these are not distributed equally. It's useful to learn extremes on both ends. In the center distance between Ps is larger.

Reply:
It looks overkill to have 3 buckets in 1-10 and 90-99 ranges to get a high level understanding of utilisation. If readability is important then we should consider putting the histogram only in event listener event and query info json, whereas the EXPLAIN should show simpler statistics like min, max, avg, stddev, median.
This hint is not going to be easily understood by most users since "buffer utilisation" is not a concept that we have documented or explained anywhere.
I'm not convinced that we need to add an English sentence about this one specific thing while all the other things in the plan are expressed in metrics. If we add this, then why not also "Join is badly ordered", or a "Statistics are missing" or "Scan is too slow" or "Stage is skewed" or all the other kind of problems a query plan might have ?
I would like to hear other opinions about whether we should head in this direction of adding diagnostic messages and how far we should take it in EXPLAIN ANALYZE VERBOSE instead of sticking to the existing practise of printing metrics.
fyi: @trinodb/maintainers

Reply:
> It looks overkill to have 3 buckets in 1-10 and 90-99 ranges to get a high level understanding of utilisation. If readability is important then we should consider putting the histogram only in event listener event and query info json, whereas the EXPLAIN should show simpler statistics like min, max, avg, stddev, median.

We don't always have access to event listener or query JSON. For example, people might be more willing to just share `EXPLAIN`. Also it's easy to explain what how to collect `EXPLAIN` to less technical user.

`min, max, avg, stddev, median.` are definitely not sufficient to derive much about distribution. Note that we already have these in basic `EXPLAIN` but even power users don't really know how to interpret these. Hence histograms are much more useful.

I consider `EXPLAIN ANALYZE VERBOSE` to be useful also for power users (like us) because it makes it easier to interpret data without browsing though query JSONS which are not very easy to read.

> This hint is not going to be easily understood by most users since "buffer utilisation" is not a concept that we have documented or explained anywhere.

I could also just say: "Output buffer is blocking" or maybe "Output buffer is a bottleneck".

>  If we add this, then why not also "Join is badly ordered", or a "Statistics are missing" or "Scan is too slow" or "Stage is skewed" or all the other kind of problems a query plan might have ?

"Stage is skwed" is added in https://github.com/trinodb/trino/pull/15286. 
I think these hints are useful for less advanced users to understand where the issues with plan are. It's potentially less work for us then since users can do initial diagnostics themselves without understanding advanced metrics.
Even for adv users like us it's easier (initially) to look for plain message rather than parse metric to quickly narrow down perf issues.

> "Scan is too slow", Join is badly ordered",

That would be super useful too. Unfortunately it's not that easy to derive these. Currently we have to do deep dive into various metrics in order to figure that either of these might be an issue. It does cost time.